pkg/scheduler/queue/batched_test.go

```diff
@@ -0,0 +1,183 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queue
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// TestBatchedProcessingPlacementSchedulingQueue_BasicOps tests the basic ops
+// (Add, Next, Done) of a batchedProcessingPlacementSchedulingQueue.
+func TestBatchedProcessingPlacementSchedulingQueue_BasicOps(t *testing.T) {
+	bq := NewBatchedProcessingPlacementSchedulingQueue("TestOnly", nil, nil, 0)
+	bq.Run()
+
+	keysToAdd := []PlacementKey{"A", "B", "C", "D", "E"}
+	for _, key := range keysToAdd {
+		bq.Add(key)
+	}
+
+	keysRecved := []PlacementKey{}
+	for i := 0; i < len(keysToAdd); i++ {
+		key, closed := bq.NextPlacementKey()
+		if closed {
+			t.Fatalf("Queue closed unexpected")
+		}
+		keysRecved = append(keysRecved, key)
+		bq.Done(key)
+		bq.Forget(key)
+	}
+
+	if !cmp.Equal(keysToAdd, keysRecved) {
+		t.Fatalf("Received keys %v, want %v", keysRecved, keysToAdd)
+	}
+
+	bq.Close()
+}
+
+// TestBatchedProcessingPlacementSchedulingQueue_BatchedOps tests the batched ops
+// (AddBatched) of a batchedProcessingPlacementSchedulingQueue.
+func TestBatchedProcessingPlacementSchedulingQueue_BatchedOps(t *testing.T) {
+	bq := NewBatchedProcessingPlacementSchedulingQueue("TestOnly", nil, nil, 5)
+	bq.Run()
+
+	addedTimestamp := time.Now()
+	keysToAddBatched := []PlacementKey{"A", "B", "C"}
+	for _, key := range keysToAddBatched {
+		bq.AddBatched(key)
+	}
+
+	keysRecved := []PlacementKey{}
+	for i := 0; i < len(keysToAddBatched); i++ {
+		key, closed := bq.NextPlacementKey()
+		if closed {
+			t.Fatalf("Queue closed unexpected")
+		}
+		keysRecved = append(keysRecved, key)
+		bq.Done(key)
+		bq.Forget(key)
+	}
+
+	if !cmp.Equal(keysToAddBatched, keysRecved) {
+		t.Fatalf("Received keys %v, want %v", keysRecved, keysToAddBatched)
+	}
+	// Allow some buffer time (1 second).
+	if timeSpent := time.Since(addedTimestamp); timeSpent < 4*time.Second {
+		t.Fatalf("time to move keys, want no less than %f seconds, got %f seconds", 4.0, timeSpent.Seconds())
+	}
+}
+
+// TestBatchedProcessingPlacementSchedulingQueue_MoveNow tests the moveNow signal
+// built in a batchedProcessingPlacementSchedulingQueue.
+func TestBatchedProcessingPlacementSchedulingQueue_MoveNow(t *testing.T) {
+	bq := NewBatchedProcessingPlacementSchedulingQueue("TestOnly", nil, nil, 10)
+	bq.Run()
+
+	keysToAddBatched := []PlacementKey{"A", "B", "C"}
+	for _, key := range keysToAddBatched {
+		bq.AddBatched(key)
+	}
+
+	// Send a move now signal.
+	bqStruct, ok := bq.(*batchedProcessingPlacementSchedulingQueue)
+	if !ok {
+		t.Fatalf("Failed to cast to batchedProcessingPlacementSchedulingQueue")
+	}
+	bqStruct.moveNow <- struct{}{}
```

Comment:
Is this intended to be a public method that the scheduler will call? If so it should  be on the public interface?